server/app/assets/javascripts/preview.ts

```diff
@@ -354,3 +362,8 @@ class PreviewController {
 export function init() {
   new PreviewController()
 }
+
+/** Safe to call multiple times */
+export function updateListeners() {
```

Comment:
non-blocking: what's the reasoning for this function? couldn't we just have callers call addOrUpdateEventListeners